Cria documentação da API

weltonfelix opened the pull request:
Cria documentação da API utilizando o padrão OpenApi 3.0 e Swagger. A visualização da documentação é feita de duas maneiras:
Redoc e Swagger UI.

O Redoc possui um visual melhor e mostra as informações de forma mais simples.

Já o Swagger UI é a forma com maior suporte, e por isso também está sendo adicionada.

weltonfelix committed:
Create API Docs

weltonfelix committed:
Merge branch 'main' into docs

weltonfelix committed:
Update swagger docs

Gvinfinity commented on docs/swagger.yaml:
```diff
@@ -0,0 +1,793 @@
+                  message:
+                    type: string
+                    description: Mensagem de erro
+                  details:
```
Retirar os detalhes para erros internos, já que não serão mais enviados a partir de #15

weltonfelix replied:
Resolved by d704c5f8ae0d03b5bc0e7fba3b00573736e2a877.

Gvinfinity commented on docs/swagger.yaml:
```diff
@@ -0,0 +1,793 @@
+      description: Acesso negado. O usuário não tem permissão para realizar esta ação.
+
+    UnauthorizedError:
+      description: Token de acesso não informado, inválido ou o usuário não tem permissão para realizar esta ação
```
Apenas Token não informado ou inválido, caso token seja válido o erro é `403 Forbidden`

weltonfelix replied:
Resolved by 3408bae0c33e25e68855851ad4df2dff7d7a22a9.

Gvinfinity commented on docs/swagger.yaml:
```diff
@@ -0,0 +1,793 @@
+            properties:
+              id:
+                type: string
+                format: studentId
```
Acrescentar explicação de como é o formato do `studentId` para clarificação

Gvinfinity replied:
Utilizar regex de validação: `/^\d{5}.{2,6}\d{4}$/g`

weltonfelix replied:
Resolved by 774834fa29ce7c328abeb8bb1fbe91cfc5649d0e.

Gvinfinity reviewed (changes requested):
Pequenas mudanças para melhoria e adequação

weltonfelix committed:
Add regex patterns and explain password requirements

weltonfelix committed:
Removes Internal Server Error response body content

Resolves https://github.com/ifpeopensource/merenda-server/pull/13\#discussion_r1112497774

weltonfelix committed:
Fix UnauthorizedError description

Resolves https://github.com/ifpeopensource/merenda-server/pull/13\#discussion_r1112498076

Gvinfinity reviewed (approved):
:rocket: :full_moon: :fireworks:
